components/datetime/src/mock.rs

```diff
@@ -124,13 +131,21 @@ impl FromStr for MockDateTime {
         let hour: IsoHour = input[11..13].parse()?;
         let minute: IsoMinute = input[14..16].parse()?;
         let second: IsoSecond = input[17..19].parse()?;
+
+        let fractional_second = if input.len() > 19 {
+            FractionalSecond::from_millisecond(input[20..23].parse()?)?
```

Comment:
suggestion: It appears that RFC 3339 allows for any arbitrary number of fractional second digits. It would be helpful for testing, if nothing else, if we allowed up to 9 digits (nanoseconds).